### ISSUE TITLE
FirebaseAdmin fails to verify Id Token when checkRevoked is set as true.

### ISSUE THREAD

**Issue by judeosbert** [appsup-dart/firebase_admin#5]
I am using FirebaseAdmin to verify the token using the following code
``` 
final authToken = request.raw.headers["token"][0];
await FirebaseAdmin.instance.app().auth().verifyIdToken(authToken);
```
When I set 
```
await FirebaseAdmin.instance.app().auth().verifyIdToken(authToken,true);
```
It returns the following error
```
FirebaseAppError(app/invalid-credential): Credential implementation provided to initializeApp() via the "credential" property failed to fetch a valid Google OAuth2 access token with the following error: "Invalid argument(s): State does not match".
```
Please check this error and let me know if I am making any mistakes.

**Comment by CosmicPangolin** [comment]
Also hit this error today.  In addition, it would be nice for cloud vm/run to work without setting a GOOGLE_CLOUD_PROJECT...I see the credential implementation is WIP :)

**Comment by Anmol92verma** [comment]
Same problem here, Can someone help ?

**Comment by akfaisel** [comment]
same error :(

**Comment by tomrozb** [comment]
Same here. Seems like this package is not maintained and I've not found other working packages for updating users using Firebase Auth.

**Comment by akfaisel** [comment]
> Same here. Seems like this package is not maintained and I've not found other working packages for updating users using Firebase Auth.

I moved on to using Node.js as backend which works seemlessly supported officially by Google docs.

I am totally new to Node.js but I have no other option :(

**Comment by rbellens** [comment]
This seems to be a problem with the authentication. How do you initialize the app? Which credentials are you using? You can get credentials by doing `Credentials.applicationDefault()` or  `Credentials.login()`.

The first one will look for credentials in the following locations:
* the env variable GOOGLE_APPLICATION_CREDENTIALS
* a configuration file, specific for this library, stored in the user's home directory
* gcloud's application default credentials
* credentials from the firebase tools

The latter will open a browser window to authenticate.

**Comment by rbellens** [comment]
I guess this is the same issue as #3, which is now fixed in version `0.1.3`.
